docs/features/software-catalog/descriptor-format.md

```diff
@@ -93,6 +93,41 @@ significance and have reserved purposes and distinct shapes.
 
 See below for details about these fields.
 
+## Substitutions In The Descriptor Format
+
+The descriptor format supports substitutions using `$text`, `$json`, and
+`$yaml`.
+
+Placeholders like `$json: https://example.com/entity.json` are substituted by
+the content of the referenced file. Files can be referenced from any configured
+integration similar to locations by passing an absolute URL. It's also possible
+to reference relative files like `./referenced.yaml` from the same location.
+There are three different types of placeholders:
+
+- `$text`: Interprets the content of the referenced file as plain text and embed
+  it as a string.
+- `$json`: Interprets the content of the referenced file as JSON and embed the
+  parsed structure.
+- `$yaml`: Interprets the content of the referenced file as YAML and embed the
+  parsed structure.
+
+For example, this can be used to load the definition of an API entity from a web
+server and embed it into the entity:
```

Comment:
```suggestion
server and embed it as a string in the field `spec.definition`:
```

(the reason i'm mentioning it like this, is that it can be a bit confusing that the `$` thing is a child inside an object, technically, but the entire object gets replaced in its parent key)